rust/gitxetcore/src/diff/csv.rs

```diff
@@ -51,8 +52,8 @@ impl SummaryDiffProcessor for CsvSummaryDiffProcessor {
         1
     }
 
-    fn get_data<'a>(&'a self, summary: &'a FileSummary) -> Option<&Self::SummaryData> {
-        summary.csv.as_ref()
+    fn get_data<'a>(&'a self, summary: &'a FileSummary) -> Option<Cow<Self::SummaryData>> {
```

Comment:
It feels like the Copy on Write mechanic here is overkill -- what's the need to use that rather than just a reference and then an explicit clone() if a writeable version is needed?

Reply:
Ah, I see...  I kinda wonder if it would just be better to use Arc<...> inside the struct, with get_data() returning Arc<..>?  I feel like the ownership model of Cow is a bit weird here, as once created we don't really intend to write it (unless I'm wrong there).

Reply:
Chatted offline about this; going with Cow for now.  A better model may be available in the future.